File: .API/Models/AssetUploadData.cs
// Decompiled with JetBrains decompiler
// Type: CloudX.Shared.AssetUploadData
// Assembly: CloudX.Shared, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
// MVID: 2635624C-5F24-4EFB-ACD1-7E9C1349B2F5
// Assembly location: F:\SteamLibrary\steamapps\common\NeosVR\HeadlessClient\CloudX.Shared.dll

using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using System.Text.Json.Serialization;

namespace CloudX.Shared
{
  public class AssetUploadData
  {
    [JsonProperty(PropertyName = "signature")]
    [JsonPropertyName("signature")]
    public string Signature { get; set; }

    [JsonProperty(PropertyName = "variant")]
    [JsonPropertyName("variant")]
    public string Variant { get; set; }

    [JsonProperty(PropertyName = "ownerId")]
    [JsonPropertyName("ownerId")]
    public string OwnerId { get; set; }

    [JsonProperty(PropertyName = "totalBytes")]
    [JsonPropertyName("totalBytes")]
    public long TotalBytes { get; set; }

    [JsonProperty(PropertyName = "chunkSize")]
    [JsonPropertyName("chunkSize")]
    public int ChunkSize { get; set; }

    [JsonProperty(PropertyName = "totalChunks")]
    [JsonPropertyName("totalChunks")]
    public int TotalChunks { get; set; }

    [JsonProperty(PropertyName = "uploadState")]
    [JsonPropertyName("uploadState")]
    [Newtonsoft.Json.JsonConverter(typeof (StringEnumConverter))]
    [System.Text.Json.Serialization.JsonConverter(typeof (JsonStringEnumConverter))]
    public UploadState UploadState { get; set; }
  }
}

File: src/api/AssetUploadData.ts
import type { UploadState } from "./UploadState";
export class AssetUploadData {
	public Signature: string;
	public Variant: string;
	public OwnerId: string;
	public TotalBytes: number;
	public ChunkSize: number;
	public TotalChunks: number;
	public UploadState: UploadState;
	constructor($b: AssetUploadDataJSON) {
		this.Signature = $b.signature;
		this.Variant = $b.variant;
		this.OwnerId = $b.ownerId;
		this.TotalBytes = $b.totalBytes;
		this.ChunkSize = $b.chunkSize;
		this.TotalChunks = $b.totalChunks;
		this.UploadState = $b.uploadState;
	}
	toJSON(): AssetUploadDataJSON {
		return {
			signature: this.Signature,
			variant: this.Variant,
			ownerId: this.OwnerId,
			totalBytes: this.TotalBytes,
			chunkSize: this.ChunkSize,
			totalChunks: this.TotalChunks,
			uploadState: this.UploadState,
		};
	}
}
export interface AssetUploadDataJSON {
	signature: string;
	variant: string;
	ownerId: string;
	totalBytes: number;
	chunkSize: number;
	totalChunks: number;
	uploadState: UploadState;
}
